core/src/smartcontracts/isi/domain.rs

```diff
@@ -126,18 +128,22 @@ pub mod isi {
                         .cloned()
                         .collect();
 
-                    for id in keys {
-                        wsv.modify_account(account.id(), |account_mut| {
-                            if account_mut.remove_asset(&id).is_none() {
-                                error!(%id, "asset not found - this is a bug");
-                            }
-
-                            Ok(AccountEvent::Asset(AssetEvent::Deleted(id)))
-                        })?;
+                    for key_id in keys {
+                        array.push((key_id, account.id().clone()));
                     }
                 }
             }
 
+            for (key_id, account_id) in array {
+                wsv.modify_account(&account_id, |account_mut| {
+                    if account_mut.remove_asset(&key_id).is_none() {
+                        error!(%key_id, "asset not found - this is a bug");
+                    }
+
+                    Ok(AccountEvent::Asset(AssetEvent::Deleted(key_id)))
+                })?;
```

Comment:
Why do we need this change?

Reply:
Borrow checker. With a patch based world state view we would be "postponing" all modifications, but checking against the unapplied changes at the same time.